trustfall_core/src/graphql_query/query.rs

```diff
@@ -400,9 +408,11 @@ fn make_field_connection(field: &Positioned<Field>) -> Result<FieldConnection, P
             }
             Some(ParsedDirective::Fold(fold, _)) => break Some(fold),
             Some(ParsedDirective::Transform(_, pos)) => {
-                return Err(ParseError::OtherError(
-                    // TODO: do better
-                    "@transform applied to non-folded edge field".to_string(),
+                return Err(ParseError::UnsupportedDirectivePosition(
+                    "@transform".to_owned(),
+                    "Cannot transform property prior to @fold directive.\
+                    Please make sure you are using @fold before @transform."
```

Comment:
```suggestion
                    Consider adding @fold before the @transform here."
```

Reply:
Tweaked to avoid implying that `@transform` must _always_ come after `@fold`. That is _currently_ the case because we have exactly one transformation operator, but in principle in the future it doesn't have to be the case.

For example, one could imagine `@transform` being used on properties directly, like `number @transform(op: "flip_sign")` or whatever.